malcolm/modules/web/parts/websocketserverpart.py

```diff
@@ -135,15 +148,18 @@ def _on_response(self, response):
                 # Websocket is dead so we can clear the subscription key.
                 # Subsequent updates may come in before the unsubscribe, but
                 # ignore them as we can't do anything about it
-                mri = self._id_to_mri.pop(response.id, None)
-                if mri:
-                    log.info(
-                        'WebSocket Error: unsubscribing from stale handle')
-                    unsubscribe = Unsubscribe(response.id)
-                    unsubscribe.set_callback(self.on_response)
-                    self._registrar.report(
-                        builtin.infos.RequestInfo(unsubscribe, mri))
-        cothread.Callback(self._queue.put, None)
+                if self._id_to_mri:
```

Comment:
Because of the class attribute being intialised to None and is therefore Optional, this gets picked up by mypy - as there is no __init__ and instead these attributes are set in initialize().

I fixed this by leaving the class attribute _id_to_mri uninitialised with a type of Dict[int, str].